pkg/sessions/redis/redis_store.go

```diff
@@ -104,11 +104,16 @@ func buildSentinelClient(opts options.RedisStoreOptions) (Client, error) {
 		return nil, err
 	}
 
+	if opts.Password != "" {
+		opt.Password = opts.Password
+	}
+
 	client := redis.NewFailoverClient(&redis.FailoverOptions{
 		MasterName:       opts.SentinelMasterName,
 		SentinelAddrs:    addrs,
 		SentinelPassword: opts.SentinelPassword,
-		Password:         opts.Password,
+		Username:         opt.Username,
```

Comment:
what happens if the parsing of the redisUrl returns an empty user? i think that could fail. therefore we need a test for the happy path with a username and a test for a test without a username.